src/run/storage/storage.cts

```diff
@@ -0,0 +1,68 @@
+// This is storage module that rest of modules should interact with.
+// Remaining modules in storage directory are implementation details
+// and should not be used directly outside of this directory.
+// There is eslint `no-restricted-imports` rule to enforce this.
+
+import { type BlobType } from '../../shared/blob-types.cjs'
+import { getTracer } from '../handlers/tracer.cjs'
+
+import { getRegionalBlobStore } from './regional-blob-store.cjs'
+import { getRequestScopedInMemoryCache } from './request-scoped-in-memory-cache.cjs'
+
+const encodeBlobKey = async (key: string) => {
+  const { encodeBlobKey: encodeBlobKeyImpl } = await import('../../shared/blobkey.js')
+  return await encodeBlobKeyImpl(key)
+}
+
+export const getMemoizedKeyValueStoreBackedByRegionalBlobStore = (
+  ...args: Parameters<typeof getRegionalBlobStore>
+) => {
+  const store = getRegionalBlobStore(...args)
+  const tracer = getTracer()
+
+  return {
+    async get<T extends BlobType>(key: string, otelSpanTitle: string): Promise<T | null> {
+      const inMemoryCache = getRequestScopedInMemoryCache()
+
+      const memoizedValue = inMemoryCache.get(key)
+      if (typeof memoizedValue !== 'undefined') {
+        return memoizedValue as T | null | Promise<T | null>
```

Comment:
We convert `undefined` back to null here, so perhaps we could push this change upstream to simplify the interface